charts/consul/templates/gateway-resources-configmap.yaml

```diff
@@ -16,4 +16,19 @@ data:
   resources.json: |
     {{ toJson .Values.connectInject.apiGateway.managedGatewayClass.resources }}
   {{- end }}
+  {{- if (mustHas "resource-apis" .Values.global.experiments) }}
+  config.yaml: |
+    gatewayClassConfigs:
+      - apiVersion: consul.hashicorp.com/v2beta1
+        metadata:
+          name: consul-mesh-gateway
+          namespace: {{ .Release.Namespace }}
+        deployment:
+          resources:
+            {{ .Values.meshGateway.resources }}
+    meshGateways:
+      - name: mesh-gateway
+        spec:
+          gatewayClassName: consul-mesh-gateway
```

Comment:
is this needed to eventually consolidate the gateways?

Reply:
Yep, we're aiming to make `MeshGateway` function as closely as possible to `Gateway` without actually using the `Gateway` resource with its current limations